server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructAutomationServices.test.js

```diff
@@ -0,0 +1,412 @@
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });    
+    const res =  ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputList, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    return res.then(() => fail())
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
+    })
+  });
+
+  test("automateForwardingConstructAsync - failed to automate ForwardingConstructAsync", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    const mockError = { message: 'Something bad happened' };
+    
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => ltps);
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fc);
+    jest.spyOn(event, 'dispatchEvent').mockImplementation(() => true);
+     
+    const res =  ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputList, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    return res.then(() => {
+      return Promise.reject(mockError);
+    })
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
+    })
+  });
+
+  test("automateForwardingConstructAsync - successful when FC Port is subscription", async () => {
+    const operationServerName = '/v1/relay-operation-update';
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => ltpsForSubscription);
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fcForSubscription);
+    jest.spyOn(event, 'dispatchEvent').mockImplementation(() => true);
+    
+    const res =  await ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputListforSubscription, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    expect(res).toBe(undefined);
+  });
+
+  test("automateForwardingConstructAsync - failed to get LogicalTerminationPointListAsync when FC Port is subscription", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    const mockError = { message: 'Something bad happened' };
+
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });
+    
+    const res =  ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputListforSubscription, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    return res.then(() => fail())
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
```

Comment:
Also if you put expect only in catch block and no error is caught, the test will again test nothing.